src/main/java/edu/ie3/datamodel/utils/ContainerUtils.java

```diff
@@ -418,4 +418,236 @@ public static JointGridContainer combineToJointGrid(
     return new JointGridContainer(
         gridName, rawGrid, systemParticipants, graphicElements, subGridTopologyGraph);
   }
+
+  /**
+   * Modifies a given {@link SubGridContainer} to make it computable for power flow calculations
+   * based on its structure. This means, that the grid is modified in a way that slack nodes are
+   * added for transformers based on assumptions about the grid, as well as all other affect
+   * entities of the grid are accordingly.
+   *
+   * <p>The following modifications are made:
+   *
+   * <ul>
+   *   <li>2 winding transformer handling
+   *       <ul>
+   *         <li>high voltage nodes are marked as slack nodes
+   *         <li>high voltage nodes in the {@link RawGridElements#getNodes()} set are replaced with
+   *             the new slack marked high voltage nodes
+   *         <li>high voltage nodes as part of {@link GraphicElements#getNodeGraphics()} are
+   *             replaced with the new slack marked high voltage nodes
+   *       </ul>
+   *   <li>3 winding transformer handling
+   *       <ul>
+   *         <li>if node a is located in this grid, no changes on 3 winding transformer nodes are
+   *             made
+   *         <li>if node b or c is located in this grid, the transformers internal node is marked as
+   *             slack node and if node a is marked as slack node, this node is unmarked as slack
+   *             node
+   *         <li>if node a got unmarked as slack, the {@link RawGridElements#getNodes()} gets
+   *             adapted accordingly
+   *         <li>in any case the internal node of the transformer is added to the {@link
+   *             RawGridElements#getNodes()} set
+   *         <li>the resulting 3 winding transformer allows for either a full calculation of this
+   *             transformer model (just ignore the internal node) or a transfer to a
+   *             'close-to-T-equivalent' model that allows for a simplified calculation of the
+   *             transformer power flows
```

Comment:
I don't know, if this is not confusing people, as it is obvious, that you can do different stuff with the transformer model.

Reply:
you're right - removed it